app/controllers/events_controller.rb

```diff
@@ -283,20 +250,62 @@ def filter_application_letters(application_letters)
       application_letters
     end
 
-    # Checks if a file is valid and not empty
-    #
-    # @param [ActionDispatch::Http::UploadedFile] is a file object
-    # @return [Boolean] whether @file is a valid file
-    def is_file?(file)
-      file.respond_to?(:open) && file.respond_to?(:content_type) && file.respond_to?(:size)
+    # Returns all file names stored in the material storage of the event
+    def get_material_files(material_path)
+      if File.exists?(material_path)
+        build_files(material_path, '')
+      else
+        []
+      end
     end
 
-    # Gets all file names stored in the material storage of the event
-    #
-    # @param [Event]
-    # @return [Array of Strings]
-    def get_material_files(event)
-      material_path = event.material_path
-      File.exists?(material_path) ? Dir.glob(File.join(material_path, "*")) : []
+    def build_files(material_path, current_path)
```

Comment:
Consider using `Find` to simplify this: http://stackoverflow.com/a/15698165/2560557